yb-voyager/cmd/importDataFileBatchProducer_test.go

```diff
@@ -329,3 +373,220 @@ func TestFileBatchProducerResumeAfterAllBatchesProduced(t *testing.T) {
+	testutils.FatalIfError(t, err)
+
+	// The second row will be too large for the batch size, but the third row is small enough to fit if the second is skipped
+	fileContents := `id,val
+1, "ok"
+2, "this row is way too long to fit in the batch and should be skipped"
+3, "ok2"`
+	_, task, err := createFileAndTask(lexportDir, fileContents, ldataDir, "test_table", 1)
+	assert.NoError(t, err)
+
+	batchproducer, err := NewFileBatchProducer(task, state, scErrorHandler)
+	assert.NoError(t, err)
+
+	// First batch: should contain row 1 and row 3 (row 2 is too large and skipped, its size does not count)
+	batch, err := batchproducer.NextBatch()
+	assert.NoError(t, err)
+	assert.NotNil(t, batch)
+	assert.Equal(t, int64(2), batch.RecordCount)
+	assert.Equal(t, int64(0), batch.Number) // last batch
+
+	// Error file should contain the error for row 2
+	assertErrorFileContains(t, lexportDir, task,
+		"larger than max batch size",
+		"ROW: 2, \"this row is way too long to fit in the batch and should be skipped\"",
+	)
+}
+
+func TestFileBatchProducer_StashAndContinue_ConversionErrorDoesNotCountTowardsBatchSize(t *testing.T) {
+	// Set max batch size in bytes to a value that would only allow two small rows, but the second row will error (conversion error) and should not count towards the batch size
+	maxBatchSizeBytes := int64(25) // enough for header + one small row + one more if no error
+	ldataDir, lexportDir, state, _, err := setupExportDirAndImportDependencies(1000, maxBatchSizeBytes)
+	testutils.FatalIfError(t, err)
+	if ldataDir != "" {
+		defer os.RemoveAll(fmt.Sprintf("%s/", ldataDir))
+	}
+	if lexportDir != "" {
+		defer os.RemoveAll(fmt.Sprintf("%s/", lexportDir))
+	}
+
+	scErrorHandler, err := importdata.GetImportDataErrorHandler(importdata.StashAndContinueErrorPolicy, getErrorsParentDir(lexportDir))
+	testutils.FatalIfError(t, err)
+
+	// The second row will error (conversion error), but is large enough that if it counted, the third row would not fit
+	fileContents := `id,val
+1, "ok"
+2, "errorrow"
+3, "ok2"`
+	_, task, err := createFileAndTask(lexportDir, fileContents, ldataDir, "test_table", 1)
+	assert.NoError(t, err)
+
+	// Use a mock valueConverter that errors on row 2
+	origValueConverter := valueConverter
+	valueConverter = &mockRowErrorValueConverter{rowToError: 2}
+	t.Cleanup(func() { valueConverter = origValueConverter })
+
+	batchproducer, err := NewFileBatchProducer(task, state, scErrorHandler)
+	assert.NoError(t, err)
+
+	// First batch: should contain row 1 and row 3 (row 2 is errored and skipped, its size does not count)
+	batch, err := batchproducer.NextBatch()
+	assert.NoError(t, err)
+	assert.NotNil(t, batch)
+	assert.Equal(t, int64(2), batch.RecordCount)
+
+	// Error file should contain the error for row 2
+	assertErrorFileContains(t, lexportDir, task,
+		"ERROR: transforming line number=2",
+		"mock conversion error",
+		"ROW: 2, \"errorrow\"",
+	)
+}
+
```

Comment:
Few test case suggestions - 

1. We can also test with the log file rotator and resumption to verify that everything is directed to the correct log file directory, even during resumption.
2. Multiple file tasks for a table.

Reply:
good points. 
1. I sort of took this lightly because this already comes for free with lumberjack, but resumption test makes sense. 
2. Makes sense. 

will take this up in a separate PR! 👍